warp10/src/main/java/io/warp10/script/functions/TOHHCODE.java

```diff
@@ -43,12 +44,21 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     Object lon = stack.pop();
     Object lat = stack.pop();
     
-    if (!(lon instanceof Number) && !(lat instanceof Number)) {
+    if (!(lon instanceof Number) || !(lat instanceof Number)) {
       throw new WarpScriptException(getName() + " expects a latitude and a longitude on the stack.");
     }
-    
-    long geoxppoint = GeoXPLib.toGeoXPPoint(((Number) lat).doubleValue(), ((Number) lon).doubleValue());
-    
+
+    double latDouble = ((Number) lat).doubleValue();
+    double lonDouble = ((Number) lon).doubleValue();
+
+    long geoxppoint = GeoTimeSerie.NO_LOCATION;
+
+    if (Double.isNaN(latDouble) != Double.isNaN(lonDouble)) {
```

Comment:
Should there be an alternate version which pushed NO_LOCATION on the stack when NaN and NaN are passed for lat/lon?